common:
    lid_topic: "/unilidar/cloud"
    imu_topic: "/unilidar/imu"
    time_sync_en: false # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu:
        0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
        # This param will take effect no matter what time_sync_en is. So if the time offset is not known exactly, please set as 0.0

preprocess:
    lidar_type: 4 # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
    scan_line: 18
    blind: 2 
    timestamp_unit: 0 # 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    point_filter_num: 1
    scan_rate: 10

mapping: # 建图和定位参数
    mapping_en: false # 是否开启建图功能
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree: 360.0
    det_range: 100.0
    extrinsic_est_en: false # true: enable the online estimation of IMU-LiDAR extrinsic
    extrinsic_T: [0.007698, 0.014655, -0.00667]
    extrinsic_R: [1, 0, 0, 0, 1, 0, 0, 0, 1]

publish:
    path_en: true
    imu2robot_T: [0, 0, 0] # imu系下机器人的位置
    imu2robot_R: [1, 0, 0,
                  0, 1, 0,
                  0, 0, 1] # imu系下机器人的姿态
    scan_publish_en: true # false: close all the point cloud output
    dense_publish_en: true # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval:
        -1 # how many LiDAR frames saved in each pcd file;
        # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

max_iteration: 3
filter_size_surf: 0.08 # 建图0.08
filter_size_map: 0.05
cube_side_length: 1000
runtime_pos_log_enable: true

prior:
    prior_T: [0 ,0, 0]
    estimateGrav: true
    prior_R: [1 ,0, 0,
              0 ,1, 0,
              0 ,0, 1]
